assets/js/atomic/blocks/product-elements/image/block.js

```diff
@@ -129,9 +129,7 @@ export const Block = ( props ) => {
 };
 
 const ImagePlaceholder = () => {
-	return (
-		<img src={ PLACEHOLDER_IMG_SRC } alt="" width={ 500 } height={ 500 } />
-	);
+	return <img src={ PLACEHOLDER_IMG_SRC } alt="" />;
```

Comment:
I saw that `alt` prop was already empty before, but I'm wondering if it would be better to add some text to explicitly say it is a placeholder image instead. What do you think?

Reply:
Super-duper agreed!

Reply:
Thank you for this suggestion! I was thinking about that, but then in a test I spotted the placeholder is searched by the empty alt:

```const placeholderImage = component.getByAltText( '' );```

and assumed it may be empty for a reason. But two `aye!`s are enough to convince me! 💪 

I'll double-check there's no other logic based on that and provide it.

Reply:
Alt text added [in the commit](https://github.com/woocommerce/woocommerce-blocks/pull/7651/commits/826f1d1c96ccb05f13d99f771380643e63e6b5fc).

I considered what should be the alt text:
- what's really on the image, like "Simple landscape drawing"
- what's the purpose, like "Generic placeholder", since in this case it doesn't really matter what's there
- something else

I went for a second option since the image itself is set externally through the settings and may change over time. Also, the content is kind of redundant in this context, the purpose matters. There are even opinions that placeholder image doesn't need alt text at all (https://www.seroundtable.com/google-alt-text-small-images-23672.html - thanks @imanish003 for sharing that! 🙌 ), but I'm not convinced.

Happy to hear your opinion!

Reply:
Great work @kmanijak. I think it is great that @imanish003 mentioned the fact that placeholder images may not need an alt text and yes I just remembered that are some situations in which the alt text must be empty, but I cannot remember from the top of my mind all those situations so I went ahead looking for resources related to that.

I found this alt Decision Tree from the W3C Web Accessibility Initiative (WAI) that I think we can use to make this decision: https://www.w3.org/WAI/tutorials/images/decision-tree/. It seems that if we consider the placeholder image as being just **decorative or not intended for the user** the best solution would be to leave it empty. Here is also a resource talking about Decorative Images: https://www.w3.org/WAI/tutorials/images/decorative/, it states that "Text values for these types of images would add audible clutter to screen reader output or could distract users if the topic is different from that in adjacent text", which is something I can agree since we can have dozens of placeholder images that is going to be read by the screen readers.

Reply:
This was a very interesting discussion. After reading it, I agree that alt text would add audible clutter to screen reader users. People do not need to hear over and over about the placeholder image which is there just to keep the layout, actually. Does not add any information.

I agree with removing it. Sorry @kmanijak for the extra work. But it's a good opportunity also because I saw there were some mistakes in the implementation which we take for granted and it's a good time to inform you about them. We never use naked strings in either JS or PHP code, we always use WordPress localization functions.

Here are two resources:

* https://codex.wordpress.org/I18n_for_WordPress_Developers
* https://www.npmjs.com/package/@wordpress/i18n

Most generally, you will see that you'll often find yourself using the `__` function.


With that said, I think we should make this decision of leaving the alt text empty explicit by adding a comment in the relevant places, perhaps linking to this discussion thread. Someone else could see the empty alt text and it would be the same story all over again!